### PR TITLE
Fix breaking change 

### PR DESCRIPTION
**Pull request by younesbelkada**
Fix a breaking change in the recent release, I made a new PR as I messed up the commit history on the previous PR

cc @sayakpaul @pacman100

### CODE REVIEW

**Review by pacman100 — APPROVED** [review]
Thank you Younes! ⚡️

File: src/peft/utils/save_and_load.py [hunk 1]
```diff
@@ -136,8 +136,27 @@ def get_peft_model_state_dict(
...
         vocab_size = getattr(getattr(model, "config", None), "vocab_size", None)
         model_id = getattr(config, "base_model_name_or_path", None)
+
+        # For some models e.g. diffusers the text config file is stored in a subfolder
+        # we need to make sure we can donwload that config.
+        has_remote_config = False
+
+        if model_id is not None:
+            try:
+                has_remote_config = file_exists(model_id, "config.jon")
```
Comment (younesbelkada) [inline src/peft/utils/save_and_load.py:146]:
```suggestion
                has_remote_config = file_exists(model_id, "config.json")
```

File: src/peft/utils/save_and_load.py [hunk 2]
```diff
@@ -136,8 +136,27 @@ def get_peft_model_state_dict(
     elif save_embedding_layers == "auto":
         vocab_size = getattr(getattr(model, "config", None), "vocab_size", None)
         model_id = getattr(config, "base_model_name_or_path", None)
+
+        # For some models e.g. diffusers the text config file is stored in a subfolder
+        # we need to make sure we can donwload that config.
```
Comment (younesbelkada) [inline src/peft/utils/save_and_load.py:141]:
```suggestion
        # we need to make sure we can download that config.
```

**Comment by DreamGenX** [comment]
Hey, looks like this fix is causing issues for me:

```
Traceback (most recent call last):
  File "/root/miniconda3/envs/py3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/root/miniconda3/envs/py3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/workspace/axolotl/src/axolotl/cli/train.py", line 59, in <module>
    fire.Fire(do_cli)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/workspace/axolotl/src/axolotl/cli/train.py", line 35, in do_cli
    return do_train(parsed_cfg, parsed_cli_args)
  File "/workspace/axolotl/src/axolotl/cli/train.py", line 55, in do_train
    return train(cfg=cfg, cli_args=cli_args, dataset_meta=dataset_meta)
  File "/workspace/axolotl/src/axolotl/train.py", line 163, in train
    trainer.train(resume_from_checkpoint=resume_from_checkpoint)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 1552, in train
    return inner_training_loop(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 1953, in _inner_training_loop
    self._maybe_log_save_evaluate(tr_loss, model, trial, epoch, ignore_keys_for_eval)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 2340, in _maybe_log_save_evaluate
    self._save_checkpoint(model, trial, metrics=metrics)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 2416, in _save_checkpoint
    self.save_model(staging_output_dir, _internal_call=True)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 2927, in save_model
    self._save(output_dir)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 2999, in _save
    self.model.save_pretrained(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/peft/peft_model.py", line 216, in save_pretrained
    output_state_dict = get_peft_model_state_dict(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/peft/utils/save_and_load.py", line 146, in get_peft_model_state_dict
    has_remote_config = file_exists(model_id, "config.json")
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 118, in _inner_fn
    return fn(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/hf_api.py", line 2386, in file_exists
    get_hf_file_metadata(url, token=token)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 118, in _inner_fn
    return fn(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 1631, in get_hf_file_metadata
    r = _request_wrapper(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 385, in _request_wrapper
    response = _request_wrapper(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 408, in _request_wrapper
    response = get_session().request(method=method, url=url, **params)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/utils/_http.py", line 67, in send
    return super().send(request, *args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/requests/adapters.py", line 501, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: (ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response')), '(Request ID: 303fade0-ec4c-44e0-b39a-0dacca0454aa)')
Traceback (most recent call last):
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/urllib3/connectionpool.py", line 791, in urlopen
    response = self._make_request(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/urllib3/connectionpool.py", line 537, in _make_request
    response = conn.getresponse()
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/urllib3/connection.py", line 461, in getresponse
    httplib_response = super().getresponse()
  File "/root/miniconda3/envs/py3.10/lib/python3.10/http/client.py", line 1375, in getresponse
    response.begin()
  File "/root/miniconda3/envs/py3.10/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/root/miniconda3/envs/py3.10/lib/python3.10/http/client.py", line 287, in _read_status
```

```
Traceback (most recent call last):
  File "/root/miniconda3/envs/py3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/root/miniconda3/envs/py3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/workspace/axolotl/src/axolotl/cli/train.py", line 59, in <module>
    fire.Fire(do_cli)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/workspace/axolotl/src/axolotl/cli/train.py", line 35, in do_cli
    return do_train(parsed_cfg, parsed_cli_args)
  File "/workspace/axolotl/src/axolotl/cli/train.py", line 55, in do_train
    return train(cfg=cfg, cli_args=cli_args, dataset_meta=dataset_meta)
  File "/workspace/axolotl/src/axolotl/train.py", line 163, in train
    trainer.train(resume_from_checkpoint=resume_from_checkpoint)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 1552, in train
    return inner_training_loop(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 1953, in _inner_training_loop
    self._maybe_log_save_evaluate(tr_loss, model, trial, epoch, ignore_keys_for_eval)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 2340, in _maybe_log_save_evaluate
    self._save_checkpoint(model, trial, metrics=metrics)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 2416, in _save_checkpoint
    self.save_model(staging_output_dir, _internal_call=True)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 2927, in save_model
    self._save(output_dir)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 2999, in _save
    self.model.save_pretrained(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/peft/peft_model.py", line 216, in save_pretrained
    output_state_dict = get_peft_model_state_dict(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/peft/utils/save_and_load.py", line 146, in get_peft_model_state_dict
    has_remote_config = file_exists(model_id, "config.json")
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 118, in _inner_fn
    return fn(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/hf_api.py", line 2386, in file_exists
    get_hf_file_metadata(url, token=token)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 118, in _inner_fn
    return fn(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 1631, in get_hf_file_metadata
    r = _request_wrapper(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 385, in _request_wrapper
    response = _request_wrapper(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 408, in _request_wrapper
    response = get_session().request(method=method, url=url, **params)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/huggingface_hub/utils/_http.py", line 67, in send
    return super().send(request, *args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/requests/adapters.py", line 501, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: (ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response')), '(Request ID: 303fade0-ec4c-44e0-b39a-0dacca0454aa)')
Traceback (most recent call last):
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/urllib3/connectionpool.py", line 791, in urlopen
    response = self._make_request(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/urllib3/connectionpool.py", line 537, in _make_request
    response = conn.getresponse()
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/urllib3/connection.py", line 461, in getresponse
    httplib_response = super().getresponse()
  File "/root/miniconda3/envs/py3.10/lib/python3.10/http/client.py", line 1375, in getresponse
    response.begin()
  File "/root/miniconda3/envs/py3.10/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/root/miniconda3/envs/py3.10/lib/python3.10/http/client.py", line 287, in _read_status
```

**Comment by younesbelkada** [comment]
Hmmm @DreamGenX it seems for some reason in your script HTTP requests are failing since `file_exists` will try to fetch from the web. Let me try to figure out a fix and get back here

**Comment by DreamGenX** [comment]
Moreover, I think there might be a bigger bug lurking here. If I am reading the code directly, when called with `save_embedding_layers="auto"` (which is the default and in many places can't be overwritten), then we will only save the embeddings if the number of tokens changes. But what if someone added embedding layers to `modules_to_save`, which unfreezes them regardless of whether tokens were added or not. Then the embeddings won't get saved?

**Comment by DreamGenX** [comment]
> Hmmm @DreamGenX it seems for some reason in your script HTTP requests are failing since `file_exists` will try to fetch from the web. Let me try to figure out a fix and get back here

Just to add to this -- downloading the base model works fine, so it's strange that the file exists check would fail later on.

**Pull request merged**